### ISSUE TITLE
Python wrapper segfaults at NonbondedForce::addParticle

### ISSUE THREAD

**Issue by farwater** [openmm/openmm#718]
Found in OpenMM compiled from the official OpenMM6.1-Source. First observed as segfaults at random times while running all python scripts (including testInstallation.py). All the non-python tests (make test) pass without any issues. All the dynamic libraries loaded by the python wrapper (segfaulting) and the test binaries (working ok), as far as I could see, are identical.

Regardless of the randomness of the segfaults of python runs, GDB sessions always point to a segfault in the call of  _wrap_NonbondedForce_addParticle as follows:

Program received signal SIGSEGV, Segmentation fault.
_int_malloc (av=0x7ffff7bb5760 <main_arena>, bytes=192) at malloc.c:3489
3489    malloc.c: No such file or directory.
(gdb) bt
#0  _int_malloc (av=0x7ffff7bb5760 <main_arena>, bytes=192) at malloc.c:3489
#1  0x00007ffff7879230 in __GI___libc_malloc (bytes=192) at malloc.c:2891
#2  0x00007ffff56e8f2d in operator new(unsigned long) () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#3  0x00007ffff62cc239 in __gnu_cxx::new_allocatorOpenMM::NonbondedForce::ParticleInfo::allocate (this=0x977d70, __n=8)

```
at /usr/include/c++/4.8/ext/new_allocator.h:104
```
#4  0x00007ffff62cb7e9 in std::_Vector_base<OpenMM::NonbondedForce::ParticleInfo, std::allocator<OpenMM::NonbondedForce::ParticleInfo> >::_M_allocate (this=0x977d70, __n=8) at /usr/include/c++/4.8/bits/stl_vector.h:168
#5  0x00007ffff62caa41 in std::vector<OpenMM::NonbondedForce::ParticleInfo, std::allocator<OpenMM::NonbondedForce::ParticleInfo> >::_M_insert_aux (this=0x977d70, __position=..., __x=...) at /usr/include/c++/4.8/bits/vector.tcc:345
#6  0x00007ffff62ca21e in std::vector<OpenMM::NonbondedForce::ParticleInfo, std::allocator<OpenMM::NonbondedForce::ParticleInfo> >::push_back (this=0x977d70, __x=...) at /usr/include/c++/4.8/bits/stl_vector.h:913
#7  0x00007ffff62c8cbb in OpenMM::NonbondedForce::addParticle (this=0x977d20, charge=0.0104, sigma=0.33996695084200002,

```
epsilon=0.45772960000000001) at /home/arsenb/Downloads/OpenMM6.1-Source/openmmapi/src/NonbondedForce.cpp:117
```
#8  0x00007ffff67910a9 in _wrap_NonbondedForce_addParticle (args=<optimized out>) at src/swig_doxygen/OpenMMSwig.cxx:59586
#9  0x000000000052f936 in PyEval_EvalFrameEx ()
#10 0x000000000055c594 in PyEval_EvalCodeEx ()
#11 0x000000000052ca8d in PyEval_EvalFrameEx ()
#12 0x000000000052cf32 in PyEval_EvalFrameEx ()
#13 0x000000000055c594 in PyEval_EvalCodeEx ()
#14 0x000000000052ca8d in PyEval_EvalFrameEx ()
#15 0x000000000055c594 in PyEval_EvalCodeEx ()
#16 0x00000000005b7392 in PyEval_EvalCode ()
#17 0x0000000000469663 in ?? ()
#18 0x00000000004699e3 in PyRun_FileExFlags ()
#19 0x0000000000469f1c in PyRun_SimpleFileExFlags ()
#20 0x000000000046ab81 in Py_Main ()
#21 0x00007ffff7817ec5 in __libc_start_main (main=0x46ac3f <main>, argc=2, argv=0x7fffffffe148, init=<optimized out>,

```
fini=<optimized out>, rtld_fini=<optimized out>, stack_end=0x7fffffffe138) at libc-start.c:287
```
#22 0x000000000057497e in _start ()

**Comment by jchodera** [comment]
Can this be closed, @farwater?

**Comment by jchodera** [comment]
Can this be closed, @farwater?

**Comment by farwater** [comment]
OK. I think, it can be closed. By trial and error I found that the issue is
reproducible for certain versions of nvidia drivers. So, eventually, by
using a combination of one of the latest nvidia drivers and one of the old
pre-compiled OpenMM versions, I managed to make OpenMM run with OpenCL.
Assuming that the interface of OpenMM with OpenCL is bug-free,  the
driver-OpenCL/cuda interface is the suspect. Overall, I think, the issue
could be closed.
